test/linalg/symmetric.jl

```diff
@@ -30,8 +30,7 @@ end
     n = 10
     areal = randn(n,n)/2
     aimg  = randn(n,n)/2
-    @testset "symmetric eigendecomposition with element type $(eltya)" for eltya in
-            (Float32, Float64, Complex64, Complex128, BigFloat, Int)
+    @testset for eltya in (Float32, Float64, Complex64, Complex128, BigFloat, Int)
```

Comment:
The issue is the types are also used later in the binary ops stuff (for example). I don't think `mat * mat` is what is meant by "Core functionality" (to me that's stuff like the c-tors and `getindex`).